### DISCUSSION TITLE
Preserving band names and no data value when using rio.to_raster

### DISCUSSION THREAD

**Discussion by xjtang** [corteva/rioxarray#529] [Q&A] [answered]
I have an xarray.DataArray with 5 bands, that I want to save as a tiff. But I want to preserve the band names. The way I do this is convert the DataArray to a Dataset first. That helps preserving the band names in the tiff file. However in that case the no_data value is lost. And I can't use rio.write_crs on a Dataset. Is there a easy solution to this?

Saving this (dataarray.rio.write_nodata(0).to_dataset(dim='band')) keeps band names but not no_data.
Saving this (dataarray.rio.write_nodata(0)) keeps no_data but not band names.

Thanks.

**Comment by snowman2** [comment, marked as answer]
Does this work?
```python
dataset = dataarray.to_dataset(dim='band')
for data_var in dataset.data_vars:
    dataset[data_var].rio.write_nodata(0, inplace=True)
```
